schema/base/document.json

```diff
@@ -28,6 +28,13 @@
       "type": "string",
       "minLength": 34,
       "maxLength": 34
+    },
+    "$meta": {
```

Comment:
We decided to hardcode meta fields, so you can remove `propertyNames` and define `userId` in `properties` and `additionalProperties: false`.